### PR TITLE
315 - spacebar play pause

### PR DESCRIPTION
**Pull request by bent0b0x**
**Issue**
#315 

**Problem**
Users cannot reliably play/pause their current video by pressing the spacebar. This functionality exists in most video players and would come across as a missing feature.

**Work Done**
- Allow player container to be focused
- Have player controller toggle play state when it receives a spacebar keydown event

### CODE REVIEW

File: imports/ui/pages/player/player.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template name="player">
-  <div id="player-container" class="player-container {{ hideControls }} {{ descriptionClass }} {{ mutedClass }} {{ volumeClass }} {{ playPause }}">
+  <div id="player-container" class="player-container {{ hideControls }} {{ descriptionClass }} {{ mutedClass }} {{ volumeClass }} {{ playPause }}" tabIndex=99>
```
Comment (bent0b0x) [inline imports/ui/pages/player/player.html:2]: This allows the container itself to be focusable and in turn receive keyboard-related events

File: tests/player.test.js [hunk 2]
```diff
@@ -110,4 +127,37 @@ describe('Player:', function () {
     assert.equal(browser.getText('#button-like'), '')
     assert.equal(browser.getText('#button-dislike'), '1')
   })
+
+  it('should play/pause a video when the spacebar is pressed', () => {
+    browser.url('http://localhost:3000/play/12345?playlist=98765')
+    browser.waitForClickable('#play-pause-button')
+    browser.switchTab()
```
Comment (bent0b0x) [inline tests/player.test.js:134]: This, at least locally, gives the browser focus (if it was previously in the background, which it typically is) and as a result allows keypresses to successfully register via the `browser.keys` api.

File: imports/ui/pages/player/player.js [hunk 3]
```diff
@@ -139,6 +148,13 @@ Template.player.onDestroyed(function () {
   Meteor.clearTimeout(controlsHandler)
 })
 
+Template.player.onRendered(function () {
+  const container = this.find('#player-container')
+  if (container) {
```
Comment (jellegerbrandy) [inline imports/ui/pages/player/player.js:153]: Should we not always ahve teh #player-container avaiblable here?

File: imports/ui/pages/player/player.js [hunk 4]
```diff
@@ -96,6 +96,15 @@ Template.player.onCreated(function () {
   // Description
   this.playerState.set('showDescription', false)
 
+  this.togglePlay = () => {
+    const dict = this.playerState
```
Comment (jellegerbrandy) [inline imports/ui/pages/player/player.js:100]: ok for now, but note that we will refactor the player to have more sane and standardized state- and event management

**Review by jellegerbrandy — APPROVED** [review]
Just 2 small remarks  @bent0b0x

**Pull request merged**